community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/checkpoint/PeriodicThresholdPolicy.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.transaction.log.checkpoint;
+
+import java.time.Clock;
+
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.helpers.Service;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.logging.LogProvider;
+
+import static org.neo4j.kernel.impl.transaction.log.checkpoint.CheckPointThreshold.or;
+
+/**
+ * The {@code periodic} check point threshold policy uses the {@link GraphDatabaseSettings#check_point_interval_time}
+ * and {@link GraphDatabaseSettings#check_point_interval_tx} to decide when check points processes should be started.
+ */
+@Service.Implementation( CheckPointThresholdPolicy.class )
+public class PeriodicThresholdPolicy extends CheckPointThresholdPolicy
+{
+    public PeriodicThresholdPolicy()
+    {
+        super( "periodically", "periodic", "interval" );
+    }
+
+    @Override
+    public CheckPointThreshold createThreshold( Config config, Clock clock, LogProvider logProvider )
+    {
+        int txThreshold = config.get( GraphDatabaseSettings.check_point_interval_tx );
+        final CountCommittedTransactionThreshold countCommittedTransactionThreshold =
+                new CountCommittedTransactionThreshold( txThreshold );
+
+        long timeMillisThreshold = config.get( GraphDatabaseSettings.check_point_interval_time ).toMillis();
```

Comment:
Could you perhaps use the `CheckPointThreshold.createTreshold` method introduced in one of your previous commits here instead?

Reply:
The `CheckPointThreshold.createTreshold` method delegates to the `createThreshold` methods on the policy objects.